overlord/devicestate/handlers_bootconfig.go

```diff
@@ -78,15 +79,17 @@ func (m *DeviceManager) doUpdateManagedBootConfig(t *state.Task, _ *tomb.Tomb) e
 	if err != nil {
 		return fmt.Errorf("cannot update boot config assets: %v", err)
 	}
+
+	// set this status already to minimize wasteful redos
+	finalStatus := state.DoneStatus
 	if updated {
 		t.Logf("updated boot config assets")
 		// boot assets were updated, request a restart now so that the
 		// situation does not end up more complicated if more updates of
 		// boot assets were to be applied
-		snapstate.RestartSystem(t, nil)
+		return snapstate.FinishTaskWithRestart(t, finalStatus, restart.RestartSystem, nil)
```

Comment:
Tbh I don't like too much to have more than one possible successful return path in a function, but anyway, not important.

Reply:
I think we try to avoid that, but there's a lot of trade offs here, also state.Hold complicates what success mean. I can try to look a bit more if there's anything simple we can do

Reply:
With the current implementation, where the last return occurs just a handful of lines below, this is not a problem. But should we add more code after this `if`, it might make things less readable.

I don't have a strong opinion, I think I would like the rest to happen in an `else` block, but I'm fine with the current code too.